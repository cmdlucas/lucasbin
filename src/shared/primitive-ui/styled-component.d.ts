import 'styled-components'
import { Theme } from './theme';

declare module 'styled-components' {
    export type WebsiteDefaultTheme = Theme
}