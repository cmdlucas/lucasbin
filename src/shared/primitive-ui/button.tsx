import styled from "styled-components";

export const PrimaryButton = styled.button(props => ({
    border: "none",
    background: props.theme.main.buttonPrimary.background,
    color: props.theme.main.buttonPrimary.color,
    fontSize: "1.25em",
    textTransform: "capitalize",
    minWidth: "160px",
    textAlign: "center",
    padding: "13.5px 20px"
}))

export const SecondaryButton = styled(PrimaryButton)(props => ({
    background: props.theme.main.buttonSecondary.background,
    color: props.theme.main.buttonSecondary.color,
}))