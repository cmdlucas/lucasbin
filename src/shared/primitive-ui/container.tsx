import styled from "styled-components";

export const HMFContainer = styled.div(() => ({
    margin: "auto",
    maxWidth: "1140px",
}))
