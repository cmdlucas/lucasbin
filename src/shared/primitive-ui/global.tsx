import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    @font-face {
        font-family: Poppins;
        src: url(/fonts/poppins/poppins-v6-latin-regular.ttf);
    }
    @font-face {
        font-family: Inconsolata;
        src: url(/fonts/inconsolata/Inconsolata-Regular.ttf);
    }
    @font-face {
        font-family: CooperHewitt;
        src: url(/fonts/cooper-hewitt/CooperHewitt-Medium.ttf);
    }
    
    * {
        margin: 0px;
        font-family: Inconsolata
    }

    a, button {
        cursor: pointer;
    }
    
    #nprogress {
        pointer-events: none;
    }

    #nprogress .bar {
        background: #29d;
        position: fixed;
        z-index: 20000;
        top: 0;
        left: 0;
        width: 100%;
        height: 2px;
    }

    /* Fancy blur effect */
    #nprogress .peg {
        display: block;
        position: absolute;
        right: 0px;
        width: 100px;
        height: 100%;
        box-shadow: 0 0 10px #29d, 0 0 5px #29d;
        opacity: 1.0;

        -webkit-transform: rotate(3deg) translate(0px, -4px);
            -ms-transform: rotate(3deg) translate(0px, -4px);
                transform: rotate(3deg) translate(0px, -4px);
    }

    /* Remove these to get rid of the spinner */
    #nprogress .spinner {
        display: block;
        position: fixed;
        z-index: 20000;
        top: 4px;
        right: 15px;
    }

    #nprogress .spinner-icon {
        width: 18px;
        height: 18px;
        box-sizing: border-box;

        border: solid 2px transparent;
        border-top-color: #29d;
        border-left-color: #29d;
        border-radius: 50%;

        -webkit-animation: nprogress-spinner 400ms linear infinite;
        animation: nprogress-spinner 400ms linear infinite;
    }

    .nprogress-custom-parent {
        overflow: hidden;
        position: relative;
    }

    .nprogress-custom-parent #nprogress .spinner,
    .nprogress-custom-parent #nprogress .bar {
        position: absolute;
    }

    @-webkit-keyframes nprogress-spinner {
        0%   { -webkit-transform: rotate(0deg); }
        100% { -webkit-transform: rotate(360deg); }
    }
    @keyframes nprogress-spinner {
        0%   { transform: rotate(0deg); }
        100% { transform: rotate(360deg); }
    }
`

export const AbsolutePosition = styled.div(props => ({
    position: "absolute",
}))

export const RelativePosition = styled.div(props => ({
    position: "relative"
}))

export const FixedPosition = styled.div(props => ({
    position: "fixed"
}))

export const HrLine = styled.hr(props => ({
    border: `1px solid ${props.theme.main.borderColor}`
}))