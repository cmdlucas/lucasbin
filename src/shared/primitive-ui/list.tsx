import styled from "styled-components";

export const List = styled.ul(() => ({
    listStyleType: "none",
    padding: 0
}))

export const ListRow = styled.li(() => ({
    width: "100%"
}))