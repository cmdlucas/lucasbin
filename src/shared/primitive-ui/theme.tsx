export interface ButtonTheme {
  color: string;
  background: string;
}

export interface ThemeProperties {
  textColor: string;
  linkColor: string;
  background: string;
  borderColor: string;
  buttonPrimary: ButtonTheme;
  buttonSecondary: ButtonTheme;
}

export const lightTheme: ThemeProperties = {
  background: "#FFFFFF",
  textColor: "#212424",
  linkColor: "#0D66D0",
  borderColor: "#EDEDED",
  buttonPrimary: {
    color: "#FFFFFF",
    background: "#212424",
  },
  buttonSecondary: {
    color: "#FFFFFF",
    background: "#366DDC",
  },
};

export const darkTheme: ThemeProperties = {
  background: "#212424",
  textColor: "#F3F3F3",
  linkColor: "#CEC652",
  borderColor: "#555555",
  buttonPrimary: {
    color: "#212424",
    background: "#FFFFFF",
  },
  buttonSecondary: {
    color: "#FFFFFF",
    background: "#366DDC",
  },
};

export type ThemeType = "light" | "dark";

export interface Theme {
  type: ThemeType;
  main: ThemeProperties;
  switchTheme: () => void;
}

export const invertTheme = (themeType: ThemeType): ThemeProperties =>
  themeType === "light" ? darkTheme : lightTheme;
  
export const invertThemeType = (themeType: ThemeType): ThemeType =>
  themeType === "light" ? "dark" : "light";

export const defaultTheme: Theme = {
  type: "light",
  main: lightTheme,
  switchTheme: () => {
    return;
  },
};
