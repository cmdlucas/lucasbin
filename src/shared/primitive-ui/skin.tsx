import { Theme } from './theme';

export interface GeneralSkin {
  width: string;
  backgroundColor: string;
  color: string;
  transition: string;
}

export const generalSkin = (theme: Theme): GeneralSkin => ({
  width: '100%',
  backgroundColor: theme.main.background,
  color: theme.main.textColor,
  transition:
    'background 0.3s ease-in 0.1s, color 0.4s ease-out 0s, border 0.4s ease-out 0s',
});
