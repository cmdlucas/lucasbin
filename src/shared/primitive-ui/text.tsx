import React, { FunctionComponent } from "react";
import styled from "styled-components";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Theme } from "./theme";
import Link from 'next/link';

const textLinkTheme = (theme: Theme, color: null | string = null, font: string | null = null) => ({
    color: color ?? theme.main.linkColor,
    textDecoration: "none",
    fontFamily: font ?? "Inconsolata"
})

interface TextLinkProps {
    href?: string
    name?: string
    color?: string
    font?: string
}

export const ExternalTextLink = styled.a<TextLinkProps>(props => 
    textLinkTheme(props.theme, props.color, props.font))

export const TextLink: FunctionComponent<TextLinkProps> = ({href, name, color, font, children}) => (
    <Link href={href} as={name} passHref>
        <ExternalTextLink color={color} font={font}>{children}</ExternalTextLink>
    </Link>
)

export const IsolatedText = styled.span(props => ({
    color: props.color ?? props.theme.main.textColor
}))

export const HeaderOne = styled.h2(() => ({
    fontSize: "2em"
}))

export const HeaderTwo = styled.h2(() => ({
    fontSize: "1.5em"
}))

export const HeaderThree = styled.h3(() => ({
    fontSize: "1.17em"
}))

export const FAIconText = styled(FontAwesomeIcon)(props => ({
    color: props.color ?? props.theme.main.textColor,
    height: "1em",
    verticalAlign: "text-top"
}))

export const IconText = styled(FAIconText)(() => ({
    paddingRight: "12px"
}))

export const Paragraph = styled.p(() => ({
    fontSize: "1em"
}))