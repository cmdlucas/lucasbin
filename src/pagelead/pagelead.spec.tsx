import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import PageLead, { HeaderIntro, HeaderText } from "./pagelead.ui";
import { IconText, TextLink } from "../shared/primitive-ui/text";
import { defaultTheme } from "../shared/primitive-ui/theme";
import { ThemeProvider } from "styled-components";

describe("PageLead Component", () => {
  let wrapper = shallow(<PageLead text="BLOG" />);

  it("should always render HeaderIntro component", () => {
    expect(wrapper.find(HeaderText).dive().find(HeaderIntro)).toHaveLength(1);
  });

  it("should render TextLink when supplied", () => {
    const Component = () => <TextLink href="/blog" />;
    const mountedTree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <PageLead text="BLOG" rightComponent={Component} />
        </ThemeProvider>
      )
      .toJSON();
    expect(mountedTree).toMatchSnapshot();
  });

  it("should render IconText when supplied", () => {
    wrapper = shallow(<PageLead text="BLOG" icon={["far", "newspaper"]} />);
    expect(wrapper.find(HeaderText).dive().find(IconText)).toHaveLength(1);
  });
});
