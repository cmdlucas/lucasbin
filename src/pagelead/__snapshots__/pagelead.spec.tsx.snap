// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLead Component should render TextLink when supplied 1`] = `
<div
  className="sc-ftvSup sc-papXJ sc-kDDrLX jaugmt IndTi gClGim"
  style={
    Object {
      "paddingBottom": "32px",
    }
  }
>
  <div
    className="sc-crXcEl ecrMWH"
  >
    <h3
      className="sc-eCYdqJ sc-evZas cBOZwO iNEMYx"
    >
       
      BLOG
       
    </h3>
  </div>
  <a
    className="sc-bczRLJ fqnNaQ"
    href="/blog"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
</div>
`;
